### ISSUE TITLE
Add _symbol property to Routine.

### ISSUE THREAD

**Issue by LonelyCat124** [stfc/PSyclone#2592]
Rather than the RoutineSymbol belonging to a Routine's SymbolTable, we should instead attach it to the `Routine` as a property (symbol) similar to how we do for References. This symbol should match (or just be ) the parent containers symbol. This therefore depends on #2575 and probably replaces @JulienRemy 's PR #2570 .

This will change a few things inside PSyclone:
- All references to `own_routine_symbol` will need to use this if they expect a RoutineSymbol
- `own_routine_symbol` may be no longer needed, and functions will just have a DataSymbol to their own name when required.
- fparser2.py needs to be update to understand what it should construct for Routines.

Some tests/constructs that involve a Routine without a Container above it will have issues with this implementation - if we have these we will need to work on how to fix them.
